### PR TITLE
feat: Adding calls to custom style hooks derived from context in react-tooltip components

### PR DESCRIPTION
**Pull request by GeoffCoxMSFT**
## Changes
- Updated components to call custom style hook

## Issues
- Updates #27058

### CODE REVIEW

**Comment by fabricteam** [comment]
## 📊 Bundle size report

| Package & Exports                                                                             |     Baseline (minified/GZIP) |                           PR |                                                                                                                                                                                                      Change |
| :-------------------------------------------------------------------------------------------- | ---------------------------: | ---------------------------: | ----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------: |
| <samp>react-tooltip</samp> <br /> <abbr title='bundle-size/Tooltip.fixture.js'>Tooltip</abbr> | `46.328 kB`<br />`16.212 kB` | `49.085 kB`<br />`16.724 kB` | `2.757 kB` <br />`512 B`  |

<details>
<summary>Unchanged fixtures</summary>

| Package & Exports                                                                                                                                                             |          Size (minified/GZIP) |
| ----------------------------------------------------------------------------------------------------------------------------------------------------------------------------- | ----------------------------: |
| <samp>react-alert</samp> <br /> <abbr title='bundle-size/Alert.fixture.js'>Alert</abbr>                                                                                       |  `90.375 kB`<br />`22.112 kB` |
| <samp>react-avatar</samp> <br /> <abbr title='bundle-size/Avatar.fixture.js'>Avatar</abbr>                                                                                    |  `51.346 kB`<br />`14.062 kB` |
| <samp>react-avatar</samp> <br /> <abbr title='bundle-size/AvatarGroup.fixture.js'>AvatarGroup</abbr>                                                                          |    `15.42 kB`<br />`6.167 kB` |
| <samp>react-avatar</samp> <br /> <abbr title='bundle-size/AvatarGroupItem.fixture.js'>AvatarGroupItem</abbr>                                                                  |  `67.568 kB`<br />`18.571 kB` |
| <samp>react-components</samp> <br /> <abbr title='bundle-size/ButtonProviderAndTheme.fixture.js'>react-components: Button, FluentProvider & webLightTheme</abbr>              |  `67.061 kB`<br />`18.105 kB` |
| <samp>react-components</samp> <br /> <abbr title='bundle-size/MultipleComponents.fixture.js'>react-components: Accordion, Button, FluentProvider, Image, Menu, Popover</abbr> | `205.644 kB`<br />`57.056 kB` |
| <samp>react-components</samp> <br /> <abbr title='bundle-size/ProviderAndTheme.fixture.js'>react-components: FluentProvider & webLightTheme</abbr>                            |  `37.891 kB`<br />`12.039 kB` |
| <samp>react-persona</samp> <br /> <abbr title='bundle-size/Persona.fixture.js'>Persona</abbr>                                                                                 |  `58.361 kB`<br />`16.009 kB` |
| <samp>react-portal-compat</samp> <br /> <abbr title='bundle-size/PortalCompat.fixture.js'>PortalCompatProvider</abbr>                                                         |    `6.324 kB`<br />`2.129 kB` |
| <samp>react-table</samp> <br /> <abbr title='bundle-size/DataGrid.fixture.js'>DataGrid</abbr>                                                                                 | `144.785 kB`<br />`39.447 kB` |
| <samp>react-table</samp> <br /> <abbr title='bundle-size/PrimitivesOnly.fixture.js'>Table (Primitives only)</abbr>                                                            |  `43.776 kB`<br />`12.063 kB` |
| <samp>react-table</samp> <br /> <abbr title='bundle-size/TableAsDataGrid.fixture.js'>Table as DataGrid</abbr>                                                                 |   `134.4 kB`<br />`34.514 kB` |
| <samp>react-table</samp> <br /> <abbr title='bundle-size/TableSelectionOnly.fixture.js'>Table (Selection only)</abbr>                                                         |  `82.316 kB`<br />`20.576 kB` |
| <samp>react-table</samp> <br /> <abbr title='bundle-size/TableSortOnly.fixture.js'>Table (Sort only)</abbr>                                                                   |  `81.637 kB`<br />`20.386 kB` |

</details>
<sub>🤖 This report was generated against <a href='https://github.com/microsoft/fluentui/commit/efb2274e37c0237f3a10a4630b2696a9880f6398'>efb2274e37c0237f3a10a4630b2696a9880f6398</a></sub>
<div id="bundleSizeComment9423"/>

**Comment by fabricteam** [comment]
## [Perf Analysis (`@fluentui/react-components`)](https://github.com/microsoft/fluentui/wiki/Perf-Testing)
<p>No significant results to display.</p><details><summary>All results</summary><p>
  <table>
  <tr>
    <th>Scenario</th>
    <th>Render type</th>
    <th>
      <a href="https://github.com/microsoft/fluentui/wiki/Perf-Testing#why-are-results-listed-in-ticks-instead-of-time-units">Master Ticks</a>
    </th>
    <th>
      <a href="https://github.com/microsoft/fluentui/wiki/Perf-Testing#why-are-results-listed-in-ticks-instead-of-time-units">PR Ticks</a>
    </th>
    <th>Iterations</th>
    <th>Status</th>
  </tr><tr>
            <td>Avatar</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/27080/perf-test-react-components/Avatar-mount_base.html">1278</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/27080/perf-test-react-components/Avatar-mount.html">1281</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Button</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/27080/perf-test-react-components/Button-mount_base.html">931</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/27080/perf-test-react-components/Button-mount.html">931</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Field</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/27080/perf-test-react-components/Field-mount_base.html">1958</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/27080/perf-test-react-components/Field-mount.html">1934</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProvider</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/27080/perf-test-react-components/FluentProvider-mount_base.html">1557</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/27080/perf-test-react-components/FluentProvider-mount.html">1558</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/27080/perf-test-react-components/FluentProviderWithTheme-mount_base.html">563</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/27080/perf-test-react-components/FluentProviderWithTheme-mount.html">583</a></td>
            <td>10</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>virtual-rerender</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/27080/perf-test-react-components/FluentProviderWithTheme-virtual-rerender_base.html">541</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/27080/perf-test-react-components/FluentProviderWithTheme-virtual-rerender.html">555</a></td>
            <td>10</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>virtual-rerender-with-unmount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/27080/perf-test-react-components/FluentProviderWithTheme-virtual-rerender-with-unmount_base.html">575</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/27080/perf-test-react-components/FluentProviderWithTheme-virtual-rerender-with-unmount.html">578</a></td>
            <td>10</td>
            <td></td>
           </tr>
<tr>
            <td>InfoButton</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/27080/perf-test-react-components/InfoButton-mount_base.html">515</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/27080/perf-test-react-components/InfoButton-mount.html">515</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>MakeStyles</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/27080/perf-test-react-components/MakeStyles-mount_base.html">1972</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/27080/perf-test-react-components/MakeStyles-mount.html">1947</a></td>
            <td>50000</td>
            <td></td>
           </tr>
<tr>
            <td>Persona</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/27080/perf-test-react-components/Persona-mount_base.html">2829</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/27080/perf-test-react-components/Persona-mount.html">2797</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>SpinButton</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/27080/perf-test-react-components/SpinButton-mount_base.html">2292</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/27080/perf-test-react-components/SpinButton-mount.html">2272</a></td>
            <td>5000</td>
            <td></td>
           </tr></table></p></details>

<div id="perfComment9425"/>

**Comment by fabricteam** [comment]
## 🕵 fluentuiv9 No visual regressions between this PR and main
<div id="vrtCommentfluentuiv9"/>

File: change/@fluentui-react-tooltip-2a113c58-5352-438c-8cdc-860e4357ad43.json [hunk 1]
```diff
@@ -0,0 +1,7 @@
+{
+  "type": "minor",
+  "comment": "feat: custom styles",
```
Comment (khmakoto) [inline change/@fluentui-react-tooltip-2a113c58-5352-438c-8cdc-860e4357ad43.json:3]:
```suggestion
  "comment": "feat: Adding calls to custom styles hooks derived from context.",
```

File: packages/react-components/react-tooltip/src/components/Tooltip/Tooltip.tsx [hunk 2]
```diff
@@ -4,6 +4,7 @@ import { renderTooltip_unstable } from './renderTooltip';
 import { useTooltipStyles_unstable } from './useTooltipStyles';
 import type { TooltipProps } from './Tooltip.types';
 import type { FluentTriggerComponent } from '@fluentui/react-utilities';
+import { useCustomStyleHooks_unstable } from '@fluentui/react-shared-contexts';
```
Comment (khmakoto) [inline packages/react-components/react-tooltip/src/components/Tooltip/Tooltip.tsx:7]: _nit:_ Keep `import type` statements at the bottom

```suggestion
import { useCustomStyleHooks_unstable } from '@fluentui/react-shared-contexts';
import { useTooltipStyles_unstable } from './useTooltipStyles';
import type { TooltipProps } from './Tooltip.types';
import type { FluentTriggerComponent } from '@fluentui/react-utilities';
```

**Review by khmakoto — APPROVED** [review]

**Pull request merged**